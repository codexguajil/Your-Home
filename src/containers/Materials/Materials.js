import React from 'react';
import { Collapse } from 'antd';
import NewMaterial from '../NewMaterial/NewMaterial';
import Material from '../Material/Material';

class Materials extends React.Component {
  state = { expanded: false };

  renderMaterialTypes = () => {
    const { materials } = this.props
    const types = Object.keys(materials)    
    return types.map((type, i) => {
      const Panel = Collapse.Panel;
      return(
        <Panel header={type} key={i}>
          {this.renderMaterials(type)}
        </Panel>
      )
    })
  }

  renderMaterials = (type) => {
    const { materials } = this.props
    return materials[type].map(material => {
      return(
        <Material {...material} type={type} key={material.id} forceRender={this.forceRender} />
      )
    });
  }

  forceRender = () => {
    this.setState({ expanded: this.state.expanded })
  }

  render() {
    const { roomId } = this.props
    return (
      <div>
        <div>
          <Collapse>
            {this.renderMaterialTypes()}
          </Collapse>
        </div>
        <div className='add-material-btn'>
          <NewMaterial roomId={roomId}
                       forceRender={this.forceRender}
          />
        </div>
      </div>
    );
  }
}

export default Materials;