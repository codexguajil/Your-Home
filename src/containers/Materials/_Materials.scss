.material {
  display: flex;
  justify-content: space-between;

  .edit-material-btn {
    margin-right: 10px;
  }
}

.add-material-btn {
  margin-top: 50px;
}

.material-control {
  display: flex;
  align-items: center;
  position: relative;
  bottom: 6px;
}