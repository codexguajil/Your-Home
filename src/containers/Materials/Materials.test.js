import React from 'react';
import { shallow } from 'enzyme';
import Materials from './Materials';


describe('Materials', () => {
  let wrapper;
  let mockMaterials;
  
  beforeEach(() => {
    mockMaterials = {
      shower: [
        {
          brand: "Kenmoore",
          id: 1,
          manual_url: null,
          model_number: "abc1",
          name: "Material 1",
          notes: null,
          quantity: null,
          unit_price: null,
          vendor: "HD"
        },
        {
          brand: "Best brand",
          id: 2,
          manual_url: null,
          model_number: "1234aa",
          name: "Material 2",
          notes: null,
          quantity: null,
          unit_price: null,
          vendor: "HD"
        }
      ],
      sink: [
        {
          brand: "Kenmoore",
          id: 3,
          manual_url: null,
          model_number: "hejso125",
          name: "Material 3",
          notes: null,
          quantity: null,
          unit_price: null,
          vendor: "HD"
        }
      ]
    }
    wrapper = shallow(<Materials materials={mockMaterials} />)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should collect all categories when renderMaterialTypes is called', () => {
    let result = wrapper.instance().renderMaterialTypes();
    expect(result.length).toBe(2);
  });

  it('should render materials when renderMaterials is called', () => {
    let result = wrapper.instance().renderMaterials('shower');
    expect(result.length).toBe(2);
  });

  it('should update state to be the same for purposes of forcing a rerender', () => {
    wrapper.setState({ expanded: false });
    let initialState = wrapper.state;
    wrapper.instance().forceRender();
    expect(wrapper.state).toBe(initialState);
  })
});