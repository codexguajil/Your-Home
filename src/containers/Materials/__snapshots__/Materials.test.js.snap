// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Materials should match snapshot 1`] = `
<div>
  <div>
    <Collapse
      bordered={true}
      expandIconPosition="left"
      openAnimation={
        Object {
          "appear": [Function],
          "enter": [Function],
          "leave": [Function],
        }
      }
    >
      <CollapsePanel
        header="shower"
        key="0"
      >
        <Material
          brand="Kenmoore"
          id={1}
          key="0"
          manual_url={null}
          model_number="abc1"
          name="Material 1"
          notes={null}
          quantity={null}
          unit_price={null}
          vendor="HD"
        />
        <Material
          brand="Best brand"
          id={2}
          key="1"
          manual_url={null}
          model_number="1234aa"
          name="Material 2"
          notes={null}
          quantity={null}
          unit_price={null}
          vendor="HD"
        />
      </CollapsePanel>
      <CollapsePanel
        header="sink"
        key="1"
      >
        <Material
          brand="Kenmoore"
          id={3}
          key="0"
          manual_url={null}
          model_number="hejso125"
          name="Material 3"
          notes={null}
          quantity={null}
          unit_price={null}
          vendor="HD"
        />
      </CollapsePanel>
    </Collapse>
  </div>
  <div
    className="add-material-btn"
  >
    <NewMaterial />
  </div>
</div>
`;
